lib/get-group-tangle.js

```diff
@@ -0,0 +1,95 @@
+// SPDX-FileCopyrightText: 2022 Mix Irving
+//
+// SPDX-License-Identifier: LGPL-3.0-only
+
+const { isCloakedMsg: isGroup } = require('ssb-ref')
+const pull = require('pull-stream')
+const Reduce = require('@tangle/reduce')
+const Strategy = require('@tangle/strategy')
+
+// for figuring out what "previous" should be for the group
+
+const strategy = new Strategy({})
+
+module.exports = function GetGroupTangle(server) {
+  const getUpdates = GetUpdates(server)
+
+  return function getGroupTangle(groupId, cb) {
+    if (!isGroup(groupId))
+      return cb(
+        new Error(`get-group-tangle expects valid groupId, got: ${groupId}`)
+      )
+
+    server.box2.getGroupKeyInfo(groupId, (err, info) => {
+      if (err) return cb(err)
+
+      if (!info)
+        return cb(new Error(`get-group-tangle: unknown groupId ${groupId}`))
+
+      getUpdates(info.root, (err, msgs) => {
+        if (err) return cb(err)
+
+        const nodes = msgs.map((msg) => ({
+          key: msg.key,
+          previous: msg.value.content.tangles.group.previous,
+        }))
+        // NOTE: getUpdates query does not get root node
+        nodes.push({ key: info.root, previous: null })
+
+        // Create a Reduce using the message contents
+        // NOTE - do NOT store the whole msg (node)
+        // we're not doing any reducing of transformations, we care only about
+        // reducing the graph to find the tips
+        // each node should be pruned down to e.g. { key: '%D', previous: ['%B', '%C'] }
+
+        const reduce = new Reduce(strategy, { nodes })
+        cb(null, {
+          root: info.root,
+          previous: Object.keys(reduce.state),
+        })
+      })
+      //)
+    })
+  }
+}
+
+const B_VALUE = Buffer.from('value')
+const B_CONTENT = Buffer.from('content')
+const B_TANGLES = Buffer.from('tangles')
+const B_ROOT = Buffer.from('root')
+
+function GetUpdates(ssb) {
+  const { seekKey } = require('bipf')
+  //const { spec } = crut
+  //const isUpdate = (msg) =>
+  //  crut.isUpdate(getCanonicalContent(msg, spec.getTransformation))
```

Comment:
```suggestion
```

No dead code

Reply:
@staltz even if this is code we might want to add in the future? i commented because it was complicated to get to work, but this is i think a safety check that existed in tribes1

Reply:
Rather make actual github issues, instead

Reply:
Removed comments and added issue https://github.com/ssbc/ssb-tribes2/issues/6